### PR TITLE
updated pre/post conditions for execute_issue

### PR DESCRIPTION
**Pull request by theodorebugnet**
* Moved the `executor` constraint for underpayment into preconditions
* Removed minimum amount precondition (NOTE: if there is in fact a minimum execution amount > 1 sat, this must be amended)
* More accurate values specified for griefing collateral, issued amounts, and issue status update

### CODE REVIEW

File: interbtc-spec/docs/source/spec/issue.rst [hunk 1]
```diff
@@ -215,14 +214,18 @@ Specification
...
             * There MUST exist a :ref:`refund-protocol` request which references ``issueId``.
 
-* The Vault's ``toBeIssuedTokens`` MUST decrease by the total amount.
-* The Vault's ``issuedTokens`` MUST increase by the total amount.
-* The user MUST receive interBTC in it's free balance.
+* The ``issue.status`` MUST be set to ``Completed``.
+* The Vault's ``toBeIssuedTokens`` MUST decrease by ``issue.amount + issue.fee``.
+* The Vault's ``issuedTokens`` MUST increase by ``issue.amount + issue.fee``.
+* The user MUST receive ``issue.amount`` interBTC in its free balance.
+* The Vault MUST receive ``issue.fee`` interBTC in its free balance.
```
Comment (gregdhill) [inline interbtc-spec/docs/source/spec/issue.rst:228]: This is not correct, it is sent to the rewards pool.

File: interbtc-spec/docs/source/spec/issue.rst [hunk 2]
```diff
@@ -194,18 +194,17 @@ Specification
 * The issue request for ``issueId`` MUST NOT have expired.
 * The ``rawTx`` MUST be valid and contain a payment to the Vault.
 * The ``rawMerkleProof`` MUST be valid and prove inclusion to the main chain.
-* The payment amount MUST include the fee calculated in :ref:`requestIssue`.
+* Either the amount transferred MUST be greater or equal to ``issue.amount + issue.fee``, or the ``executor`` MUST be the account that made the issue request.
```
Comment (gregdhill) [inline interbtc-spec/docs/source/spec/issue.rst:197]: Might make sense to split this into separate if conditions?

**Review by gregdhill — CHANGES_REQUESTED** [review]

File: interbtc-spec/docs/source/spec/issue.rst (hunk 1 above)
Comment (theodorebugnet) [reply interbtc-spec/docs/source/spec/issue.rst:228]: What's the best way to specify this? `The distributeRewards function MUST be called with issue.fee`, or `the fee pool MUST increase by issue.fee`?

Comment (gregdhill) [reply interbtc-spec/docs/source/spec/issue.rst:228]: I think the latter for now.

**Comment by gregdhill** [comment]
Please also make the update from here: https://github.com/interlay/interbtc-spec/pull/23

**Review by gregdhill — APPROVED** [review]

**Pull request merged**